pytng/pytng.pyx

```diff
@@ -234,31 +276,31 @@ cdef class TNGFile:
             self.reached_eof = True
             raise StopIteration("Reached EOF in read")
 
-        cdef MemoryWrapper wrap
-        cdef float* positions
-        wrap = MemoryWrapper(3 * self.n_atoms * sizeof(float))
-        positions = <float*> wrap.ptr
-
-        cdef int64_t stride_length, ok, i, n_values_per_frame
-        ok = tng_util_pos_read_range(self._traj, self.step, self.step, &positions, &stride_length)
+        cdef MemoryWrapper wrap_pos
+        cdef float* position
+        wrap_pos = MemoryWrapper(3 * self.n_atoms * sizeof(float))
+        positions = <float*> wrap_pos.ptr
+        cdef int64_t stride_length, ok
+        ok = tng_util_pos_read_range(self._traj, self.step, self.step, &positions, &stride_length) #TODO this will break when using frames spaced more than 1 apart
```

Comment:
Did you add a test with frames > 1 to break it? That might be a start (perhaps with XFAIL(.

Reply:
working on it now :)

Reply:
This hinges on what we mean by specifically by "frames". As currently stands there is nothing wrong with the current code other than that  `n_frames` is the number of md steps.